spanner/value.go

```diff
@@ -3564,6 +3614,24 @@ func encodeValue(v interface{}) (*proto3.Value, *sppb.Type, error) {
 		pt = proto.Clone(v.Type).(*sppb.Type)
 	case []GenericColumnValue:
 		return nil, nil, errEncoderUnsupportedType(v)
+	case protoreflect.Enum:
+		var protoEnumfqn string
+		if v != nil {
+			pb.Kind = stringKind(strconv.FormatInt(int64(v.Number()), 10))
+			protoEnumfqn = string(v.Descriptor().FullName())
+		}
+		pt = enumType(protoEnumfqn)
+	case proto.Message:
+		var protoMessagefqn string
+		if v != nil && proto.MessageReflect(v).IsValid() {
+			bytes, err := proto.Marshal(v)
+			if err != nil {
+				return nil, nil, err
+			}
+			pb.Kind = stringKind(base64.StdEncoding.EncodeToString(bytes))
+			protoMessagefqn = string(proto.MessageReflect(v).Descriptor().FullName())
+		}
+		pt = protoType(protoMessagefqn)
```

Comment:
Same as above.

Reply:
If `v` is nil, then `protoMessagefqn` by default gets assigned with default value of string which is `""`.

Reply:
Added unit test for this also. 
As per protoreflect documentation, `(*pb.SingerInfo)(nil)` is an invalid message type. So setting `protoMessagefqn` as `""` is valid.

Text from Protoreflect document:
`An invalid message often corresponds to a nil pointer of the concrete message type`